src/library.js

```diff
@@ -1086,7 +1086,7 @@ LibraryManager.library = {
       var info = EXCEPTIONS.infos[ptr];
       assert(info.refcount > 0);
       info.refcount--;
-      if (info.refcount === 0) {
+      if (info.refcount === 0 && ! info.rethrown) {
```

Comment:
please remove the space after the `!`

also please add a comment explaining what you said in a comment in the PR, that a rethrown exception can get to 0, must be kept alive, and will later be incremented up and back down to 0 to be actually cleaned up.